### PR TITLE
feat: Run test coverage and upload artifacts to codecov

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
Create coverage test artifacts and uplooad to codecov. The results can be used by https://shields.io/category/coverage to get the coverage percentage

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18626
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by flacatus** [comment]
cc @rhopp

**Comment by flacatus** [comment]
Example in my PR:
image

**Comment by tolusha** [comment]
@flacatus 
Can we have that badge in our README.md file ?

@rhopp 
Is it or for you?

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Install chectl dependencies
+        run: yarn
+      - name: Install codecov
+        run: npm install codecov -g
+      - name: Run unit tests
+        run: yarn test
+      - name: Upoad coverage test results
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        run: bash <(curl -s https://codecov.io/bash)
```
Comment (tolusha) [inline .github/workflows/codecov.yml:24]:
```suggestion
name: chectl
on: [push]
jobs:
  build-and-deploy:
    runs-on: ubuntu-20.04
    name: Code coverage
    steps:
      - name: Checkout
        uses: actions/checkout@v2.3.1
        with:
          persist-credentials: false
      - uses: actions/setup-node@v1
        with:
          node-version: '12'
      - name: Install chectl dependencies
        run: yarn
      - name: Install codecov
        run: npm install codecov -g
      - name: Run unit tests
        run: yarn test
      - name: Upload code coverage results
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
        run: bash <(curl -s https://codecov.io/bash)
```

**Comment by rhopp** [comment]
Yep. Readme is fine ;-)

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
README.md is not up to date ^(

**Comment by flacatus** [comment]
> README.md is not up to date ^(

Fixed.

**Pull request merged**